src/main/kotlin/com/example/delivery/controller/ProductController.kt

```diff
@@ -16,30 +16,31 @@ import org.springframework.web.bind.annotation.RequestBody
 import org.springframework.web.bind.annotation.RequestMapping
 import org.springframework.web.bind.annotation.ResponseStatus
 import org.springframework.web.bind.annotation.RestController
+import reactor.core.publisher.Mono
 
 @RestController
 @RequestMapping("/products")
 class ProductController(private val productService: ProductService) {
     @LogInvoke
     @GetMapping("/{id}")
-    fun findById(@PathVariable id: String): ProductDTO {
-        return productService.getById(id).toDTO()
+    fun findById(@PathVariable id: String): Mono<ProductDTO> {
```

Comment:
Do you return 404 when not found? If so, where do you map Mono.empty to 404 Http status?

Reply:
Yes, i return 404 because i handle this exception in GlobalExceptionHandler